src/index.ts

```diff
@@ -51,6 +52,17 @@ export class TestxServer {
     };
 
     const apolloServer = new ApolloServer({ typeDefs, resolvers, context });
-    return apolloServer;
+
+    // Sets PORT as a random port available in the range 29170 - 29998
+    const ports = await portastic.find({ min: 29170, max: 29998 });
+    const PORT = ports[Math.floor(Math.random() * 100)];
```

Comment:
There is a chance that for example, 90 ports are available and that the random function returns 98, in this case, `port` will become `undefined`, in my opinion, is easier to just pick the first and throw an error if no ports are available in the range:
```js
if (ports.length< 1) { 
    throw new Error(`no free ports available in range 29170 - 29998`) 
}
const port = ports[0];
```